### DISCUSSION TITLE
Does GitHub secret scanning detect BIP-39 mnemonics?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [community/community#59751] [Code Security] [answered]
### Select Topic Area

Question

### Body

I'm researching GitHub's secret scanning functionality, and I'm interested to know if GitHub looks for [BIP-39 mnemonics](https://iancoleman.io/bip39/) when it scans the repo for leaked secrets?

**Comment by aashah** [comment, marked as answer]
We do not. The full list of supported patterns can be found here: https://docs.github.com/en/code-security/secret-scanning/secret-scanning-patterns#supported-secrets

Secrets without clear detectable patterns are difficult to scan, as we look at **all** git content that flows through GitHub. Without clear ownership as well, it's difficult to properly remediate the leak.

If the pattern for BIP-39 is strong, and can be represented in both a regex and through some psuedocode to validate it, it may be something we can explore.
